### PR TITLE
compilation: force colored cargo output

### PR DESCRIPTION
**Pull request by birkenfeld**
Alternative to #397 which works for any invocation of `compile`.

### CODE REVIEW

**Comment by brotzeit** [comment]
If somebody else can test this I would be happy to merge this PR.

**Comment by brotzeit** [comment]
@mookid what do you think ?

File: rust-mode.el [hunk 1]
```diff
@@ -16,7 +16,8 @@
 
 (eval-when-compile (require 'rx)
                    (require 'compile)
-                   (require 'url-vars))
+                   (require 'url-vars)
+                   (require 'ansi-color))
```
Comment (nbfalcon) [inline rust-mode.el:20]: `ansi-color-apply-on-region` is a function, not a macro, so shouldn't this *not* be in `eval-when-compile`?

**Comment by psibi** [comment]
I'm gently closing this PR for now as there haven't been any recent updates. However, feel free to reopen it whenever you're ready to pick it up again.

**Pull request closed without merging**